src/OICodeGen.cpp

```diff
@@ -2976,8 +2968,10 @@ bool OICodeGen::generateJitCode(std::string &code) {
       } else {
         LOG(ERROR) << "Failed to get size: " << err->code << " " << err->message
                    << " " << e;
+        drgn_error_destroy(err);
         return false;
       }
+      drgn_error_destroy(err);
```

Comment:
Would it be clearer if this was in the positive if branch?

Reply:
Actually the way I did it sucks. I've rewritten it with a boolean that is set to true if we should return and then checked below. This allows the error cleanup to be called just once and it's much tidier.